gnovm/cmd/gno/test.go

```diff
@@ -328,9 +328,10 @@ func gnoTestPkg(
 
 		// tfiles, ifiles := gno.ParseMemPackageTests(memPkg)
 		tfiles, ifiles := parseMemPackageTests(memPkg)
+		testPkgName := getPkgNameFromFileset(ifiles)
 
 		// run test files in pkg
-		{
+		if !strings.HasSuffix(testPkgName, "_test") {
```

Comment:
The problem with this approach is that a package can only contain either files declared as `package pkg_test` or as `package pkg`. To see this, in `examples/gno.land/p/demo/tests`:

```console
$ cat << EOF > a_test.gno
package tests

import "testing"

func TestHello2(t *testing.T) {
        t.Log("hello!")
}
EOF
$ gno test -verbose .
=== RUN   TestGetHelloWorld
hello world
--- PASS: TestGetHelloWorld (0.00s)
=== RUN   file/z0_filetest.gno
--- PASS: file/z0_filetest.gno (0.01s)
ok      .       1.70s
```

So, aside from making it work for this one, I recommend you add a txtar tests to make sure all tests are collected and executed properly.

Reply:
Damn! just fixed!